### PR TITLE
Rework build.sbt to provide a separate midas-targetUtils package

### PR DESCRIPTION
**Pull request by davidbiancolin**
Associated PR: ucb-bar/midas#106
Here's my attempt to tease apart the annotations in MIDAS that you'd want to use in Rocket Chip RTL. Despite my earnest attempts to do things properly, SBT wasn't having it. 

I'm open to better names for the midas package. 

@alonamid

### CODE REVIEW

**Comment by alonamid** [comment]
The name makes sense to me (although that as a "utils" package, it might end up being a home to more things that are not only annotations?)

**Comment by davidbiancolin** [comment]
> The name makes sense to me (although that as a "utils" package, it might end up being a home to more things that are not only annotations?)

We're planning on added some of the MIDAS 2.0 transformations here too, specifically those that are target-rtl preserving.

**Comment by davidbiancolin** [comment]
This is ready for review, but i'll merge it in after super node goes in.

**Comment by alonamid** [comment]
Probably need to update the AutoILA docs with the correct package name

**Comment by davidbiancolin** [comment]
@alonamid @sagark  This is ready for merge. I squashed all the intermediate commits since they were busted for various reasons (rebased submodules, SBT bugs).

File: sim/build.sbt [hunk 1]
```diff
@@ -25,14 +21,62 @@ def isolateAllTests(tests: Seq[TestDefinition]) = tests map { test =>
...
+  .settings(
+    commonSettings,
+    crossScalaVersions := Seq("2.11.12", "2.12.4"))
+  .dependsOn(chisel, midasTargetUtils)
+lazy val macros     = (project in rocketChipDir / "macros")
+  .settings(commonSettings)
+
+// HACK: I'm strugging to override settings in rocket-chip's build.sbt (i want
+// the subproject to register a new library dependendency on midas's targetutils library)
+// So instead, avoid the existing build.sbt altogether and specify the project's root at src/
```
Comment (sagark) [inline sim/build.sbt]: classic

**Pull request merged**